codegen/smithy-aws-go-codegen/src/main/java/software/amazon/smithy/aws/go/codegen/JsonShapeDeserVisitor.java

```diff
@@ -190,10 +192,16 @@ protected void deserializeStructure(GenerationContext context, StructureShape sh
                     }
                     String memberName = symbolProvider.toMemberName(member);
                     String serializedMemberName = getSerializedMemberName(member);
-                    writer.openBlock("case $S:", "", serializedMemberName, () -> {
-                        String dest = "sv." + memberName;
-                        context.getModel().expectShape(member.getTarget()).accept(getMemberDeserVisitor(member, dest));
-                    });
+
+                    boolean isErrorStructure = shape.hasTrait(ErrorTrait.class);
+
+                    if(isErrorStructure && memberName.equals("Message")){
+                        writer.write("case \"message\", \"Message\":");
+                    } else {
+                        writer.write("case $S:",serializedMemberName);
+                    }
+                    String dest = "sv." + memberName;
+                    context.getModel().expectShape(member.getTarget()).accept(getMemberDeserVisitor(member, dest));
```

Comment:
This should fix the missing newline issue (I didn't check it myself though)

```suggestion
                    context.getModel().expectShape(member.getTarget()).accept(getMemberDeserVisitor(member, dest));
                    writer.write("");
```